File: palindrome-num.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
       
        temp = str(x) #String solution
        for i in range(len(temp) // 2):
            if temp[i] != temp[-(i+1)]:
                return False
        return True

File: palindrome-num.c
bool isPalindrome(int x){
    if (x < 10 && x > -1) {
        return true;
    }
    if (x < 0) {
        return false;
    }

    int temp = x;
    long final = 0;

    while (temp > 0) {
        final *= 10;
        final += temp % 10;
        temp /= 10; 
    }
    return final == x;
}
